app/models/shipit/duration.rb

```diff
@@ -0,0 +1,31 @@
+module Shipit
+  class Duration
+    UNITS = {
+      's' => :seconds,
+      'm' => :minutes,
+      'h' => :hours,
+      'd' => :days,
+    }.freeze
+
+    def initialize(seconds)
+      @seconds = seconds
+    end
+
+    def to_i
+      @seconds.to_i
+    end
+
+    def to_s
+      seconds = to_i
+      minutes, seconds = seconds.divmod(60)
+      hours, minutes = minutes.divmod(60)
+      days, hours = hours.divmod(24)
+
+      components = [days, hours, minutes, seconds].drop_while(&:zero?)
+      return "0s" if components.empty?
+
+      first, *rest = components.reverse.zip(UNITS.keys).reverse
```

Comment:
We could avoid that by doing the `drop_while` after zipping. But I think tests covering it should be enough.